### PR TITLE
Implement tasks 01-15

### PR DESCRIPTION
**Pull request by skynetplace**
Implemented and refactored tasks from 1 to 15 inclusive.

### CODE REVIEW

File: src/main/java/com/honcharenko/TwentyMethods.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    }
+
+    // 11) Принимает инт год и возвращает тру, если год высокосный.
+    public static boolean isLeapYear(int year) {
+        if (year % 400 == 0) {
+            return true;
+        } else if (year % 100 == 0) {
+            return true;
+        }
+        return year % 4 == 0;
```
Comment (BogdanParashchak) [inline src/main/java/com/honcharenko/TwentyMethods.java:120]: 1. check logic
2. chack for unnessesary 'else'

File: src/main/java/com/honcharenko/TwentyMethods.java [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    public static void filterIntArrayMultiples(int[] ints, int number) {
+        for (int element : ints) {
+            if (element % number == 0) {
+                System.out.println(element);
+            }
+        }
+    }
+
+    // 13) Метод принимает массив интов, сортирует его по возрастанию.
+    public static void bubblySortIntArrayInAscendingOrder(int[] ints) {
```
Comment (BogdanParashchak) [inline src/main/java/com/honcharenko/TwentyMethods.java:134]: bubbleSortIntArrayInAscendingOrder

File: src/main/java/com/honcharenko/TwentyMethods.java [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    // 15) Принимает два массива интов одинаковых по длинне, возращает массив интов, который состоит из перемноженных елементов входящих массивов.
+    public static int[] multiplyArrays(int[] firstIntArray, int[] secondIntArray) {
+        int[] resultIntArray = new int[firstIntArray.length];
```
Comment (BogdanParashchak) [inline src/main/java/com/honcharenko/TwentyMethods.java:175]: not needed
write result into one of input arrays

File: src/main/java/com/honcharenko/TwentyMethods.java [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+        for (int i = 0; i < firstIntArray.length; i++) {
+            resultIntArray[i] = firstIntArray[i] * secondIntArray[i];
+        }
+        return resultIntArray;
+
+    }
+
+// 16) Принимает два массива интов, возвращает массив из елементов, которые не совпадают в массивах.
```
Comment (BogdanParashchak) [inline src/main/java/com/honcharenko/TwentyMethods.java:183]: delete

File: src/main/java/com/honcharenko/TwentyMethods.java [hunk 5]
```diff
@@ -0,0 +1,187 @@
+package com.honcharenko;
+
+public class TwentyMethods {
+    // 1) Принимает массив чаров, выводит его на экран.
```
Comment (BogdanParashchak) [inline src/main/java/com/honcharenko/TwentyMethods.java:4]: replace with javadoc in the whole class

File: pom.xml
Comment (BogdanParashchak) [inline pom.xml:1]: replace with javadoc in the whole class

File: src/main/java/com/honcharenko/Starter.java
Comment (BogdanParashchak) [inline src/main/java/com/honcharenko/Starter.java:1]: replace with javadoc in the whole class

File: src/main/java/com/honcharenko/TwentyMethods.java
Comment (BogdanParashchak) [inline src/main/java/com/honcharenko/TwentyMethods.java:1]: replace with javadoc in the whole class

**Review by BogdanParashchak — CHANGES_REQUESTED** [review]
minor changes to be done

File: src/main/java/com/honcharenko/TwentyMethods.java [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+        for (char item : chars) {
+            string += item;
+        }
+        return string;
+    }
+
+    // 8) Принимает массив интов и значение типа инт.
+    // Возвращает индекс массива, в котором значение совпадает с передаваемым начиная с начала массива.
+    // Если значения в массиве нет - возвращает -1.
+    public static int searchFirstIndexOfValueInArray(int[] ints, int searchValue) {
```
Comment (diamandv) [inline src/main/java/com/honcharenko/TwentyMethods.java:72]: searchFirstIndexOfValueInArray -> findFirstIndexOfSearchingValueInArray

**Review by diamandv — COMMENTED** [review]
.

File: pom.xml [hunk 7]
```diff
@@ -5,7 +5,7 @@
     <modelVersion>4.0.0</modelVersion>
 
     <groupId>com.honcharenko</groupId>
-    <artifactId>twenty-methods</artifactId>
+    <artifactId>TwentyMethods</artifactId>
```
[the remark this answers is not in the document]
Comment (skynetplace) [reply pom.xml:8]: Done

File: src/main/java/com/honcharenko/TwentyMethods.java [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+        return chars;
+    }
+
+    // 3) Принимает 2 инта, а и б, возвращает большее из этих 2х чисел.
+    public static int getMaxInt(int firstNumber, int secondNumber) {
+        return Math.max(firstNumber, secondNumber);
+    }
+
+    // 3.1) Принимает 2 инта, а и б, возвращает большее из этих 2х чисел.
+    public static int getMaxIntByTernary(int firstNumber, int secondNumber) {
```
[the remark this answers is not in the document]
Comment (skynetplace) [reply src/main/java/com/honcharenko/TwentyMethods.java:25]: Done

File: src/main/java/com/honcharenko/TwentyMethods.java [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+        return (firstNumber >= secondNumber) ? firstNumber : secondNumber;
+    }
+
+    // 3.2) Принимает 2 инта, а и б, возвращает большее из этих 2х чисел.
+    public static int getMaxIntByBranching(int firstNumber, int secondNumber) {
```
[the remark this answers is not in the document]
Comment (skynetplace) [reply src/main/java/com/honcharenko/TwentyMethods.java:30]: Done

File: src/main/java/com/honcharenko/TwentyMethods.java [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+
+    // 4) Принимает 3 инта, возвращает большее из них.
+    public static int getMaxInt(int firstNumber, int secondNumber, int thirdNumber) {
+        if (firstNumber >= secondNumber && firstNumber >= thirdNumber) {
+            return firstNumber;
+        } else if (secondNumber >= firstNumber && secondNumber >= thirdNumber) {
+            return secondNumber;
+        } else {
+            return thirdNumber;
+        }
```
[the remark this answers is not in the document]
Comment (skynetplace) [reply src/main/java/com/honcharenko/TwentyMethods.java:45]: Done

File: src/main/java/com/honcharenko/TwentyMethods.java [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+            factorial *= i;
+        }
+        return factorial;
+    }
+
+    // 10.1) Метод принимает инт и возвращает факториал от заданого инта.
+    public static int calculateFactorialRecursive(int number) {
+        if (number == 0) {
+            return 1;
+        } else {
```
[the remark this answers is not in the document]
Comment (skynetplace) [reply src/main/java/com/honcharenko/TwentyMethods.java:108]: Done

File: src/main/java/com/honcharenko/TwentyMethods.java [hunk 12]
```diff
@@ -0,0 +1,187 @@
...
+    public static int calculateFactorial(int number) {
+        int factorial = 1;
+        for (int i = number; i > 1; i--) {
+            factorial *= i;
+        }
+        return factorial;
+    }
+
+    // 10.1) Метод принимает инт и возвращает факториал от заданого инта.
+    public static int calculateFactorialRecursive(int number) {
```
[the remark this answers is not in the document]
Comment (skynetplace) [reply src/main/java/com/honcharenko/TwentyMethods.java:105]: Done

File: src/main/java/com/honcharenko/TwentyMethods.java (hunk 1 above)
Comment (skynetplace) [reply src/main/java/com/honcharenko/TwentyMethods.java:120]: Fixed

File: src/main/java/com/honcharenko/TwentyMethods.java (hunk 2 above)
Comment (skynetplace) [reply src/main/java/com/honcharenko/TwentyMethods.java:134]: Done

File: src/main/java/com/honcharenko/TwentyMethods.java (hunk 3 above)
Comment (skynetplace) [reply src/main/java/com/honcharenko/TwentyMethods.java:175]: Done

File: src/main/java/com/honcharenko/TwentyMethods.java (hunk 4 above)
Comment (skynetplace) [reply src/main/java/com/honcharenko/TwentyMethods.java:183]: Done

File: pom.xml
Comment (skynetplace) [reply pom.xml:1]: File without comments. Done

File: src/main/java/com/honcharenko/TwentyMethods.java
Comment (skynetplace) [reply src/main/java/com/honcharenko/TwentyMethods.java:1]: Done

File: src/main/java/com/honcharenko/TwentyMethods.java (hunk 6 above)
Comment (skynetplace) [reply src/main/java/com/honcharenko/TwentyMethods.java:72]: Done

File: src/main/java/com/honcharenko/TwentyMethods.java [hunk 13]
```diff
@@ -0,0 +1,187 @@
...
+                return i;
+            }
+            i++;
+        }
+        return -1;
+    }
+
+
+    // 9) Принимает массив интов и значение типа инт. Возвращает индекс массива, в котором значение совпадает с передаваемым начиная с конца массива. Если значения в массиве нет - возвращает -1.
+    public static int searchLastIndexOfValueInArray(int[] ints, int searchValue) {
```
[the remark this answers is not in the document]
Comment (skynetplace) [reply src/main/java/com/honcharenko/TwentyMethods.java:85]: Done

File: src/main/java/com/honcharenko/TwentyMethods.java (hunk 10 above)
[the remark this answers is not in the document]
Comment (BogdanParashchak) [reply src/main/java/com/honcharenko/TwentyMethods.java]: else if -> if

File: src/main/java/com/honcharenko/TwentyMethods.java [hunk 14]
```diff
@@ -0,0 +1,218 @@
...
+        return secondNumber;
+    }
+
+    /**
+     * 4) Принимает 3 инта, возвращает большее из них.
+     */
+    public static int getMaxInt(int firstNumber, int secondNumber, int thirdNumber) {
+        if (firstNumber >= secondNumber && firstNumber >= thirdNumber) {
+            return firstNumber;
+        } else if (secondNumber >= firstNumber && secondNumber >= thirdNumber) {
```
Comment (BogdanParashchak) [inline src/main/java/com/honcharenko/TwentyMethods.java:55]: if (secondNumber >= firstNumber && secondNumber >= thirdNumber) {

**Review by BogdanParashchak — CHANGES_REQUESTED** [review]

File: src/main/java/com/honcharenko/TwentyMethods.java (hunk 14 above)
Comment (skynetplace) [reply src/main/java/com/honcharenko/TwentyMethods.java]: Fixed

**Review by diamandv — APPROVED** [review]

**Pull request merged**